src/gwc/src/main/java/org/geoserver/gwc/dispatch/GwcServiceProxy.java

```diff
@@ -173,13 +181,31 @@ public void verifyAccess(HttpServletRequest rawRequest) throws org.geotools.ows.
         } else if (rawRequest.getPathInfo().toLowerCase().startsWith("/service/gmaps") || rawRequest.getPathInfo().toLowerCase().startsWith("/service/ve")) {
             String layerstr = (String) parameters.get("LAYERS");
                         
-            if (layerstr != null) {                
+            if (layerstr != null) {
+                int level = Integer.parseInt((String) parameters.get("zoom"));
+                long col = Long.parseLong((String) parameters.get("x"));
+                long row = Long.parseLong((String) parameters.get("y"));
+                try {
+                    long[] converted = GMapsConverter.convert(level, col, row);
+                    col = converted[0];
+                    row = converted[1];
```

Comment:
Admittedly the converter doesn't do anything to the level here, but for consistency it would be best to set the zoom level based on the returned value along with `row` and `col`.  In particular that would make it clearer when adapting the code for the mgmaps protocol

I would probably have used new variables to keep the GMaps and GWC coordinates distinct for readability.